### ISSUE TITLE
Crash to desktop when worker is fired or removed from job

### ISSUE THREAD

**Issue by Oraxian** [ldtteam/minecolonies#7373]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.14.273-ALPHA
- Structurize Version: 0.13.198-ALPHA
- Related Mods and their Versions:

## Expected behavior
You should be able to fire a worker from their job at their job hut.

## Actual behavior
This behavior started when I updated from 0.14.263-ALPHA to 0.14.273-ALPHA.  

When a worker is fired from their job, the game crashes to desktop.  This happens if you do this from the "manage workers" screen at the job hut, but I have also seen it happen when a student in the school is removed because they grow up.  Hiring a worker works as expected.  I reproduced this behavior both on my long-term build, and a brand new creative test world.

Please refer to attached latest log.  A block similar to the one starting at line 1050 shows up each time a crash occurs.

## Steps to reproduce the issue

1. Update to 0.14.273-ALPHA
2. Fire a worker anywhere

### Logs

https://gist.github.com/Oraxian/81f842d42d75eecb43a61152753840b1

### Notes

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
What's your forge version?

**Comment by Oraxian** [comment]
> What's your forge version?

Forge 36.1.32

I'm playing the Minecolonies Official v2.9 modpack on CurseForge.  I update minecolonies and structurize manually as new versions are released.

**Comment by Raycoms** [comment]
Can you see if the next version fixes it?

**Comment by Oraxian** [comment]
> Can you see if the next version fixes it?

I installed version 274, and the issue still exists, both in existing and new worlds.  Here's an updated log:

https://gist.github.com/Oraxian/b61471025a6a8fdbbf95d9711c1e4fd4

Let me know if I can help in any way, and thanks for all your efforts!

**Comment by Oraxian** [comment]
Not sure if this helps, but I did a little more testing.  I created a new instance with only 2 mods installed (same Forge version:) Minecolonies (v274) and structurize (v198.)  I created a new creative world, made a town hall and builder hut, and then fired the builder at the hut block - same CTD as before.  Attaching new log below.  The screenshot below is the error from the launcher.

Here's the new log:
https://gist.github.com/Oraxian/ed5d34ae35ba087654831e962f001d45

Here's the screenshot of the CTD:
![Capture](https://user-images.githubusercontent.com/37782499/124848166-e6af4500-df61-11eb-812a-0630120f1ea7.PNG)
